server/src/main/java/io/deephaven/server/table/stats/BigDecimalChunkedNumericalStats.java

```diff
@@ -0,0 +1,106 @@
+package io.deephaven.server.table.stats;
+
+import io.deephaven.chunk.ObjectChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.configuration.Configuration;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.ChunkSource;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.impl.util.ColumnHolder;
+import io.deephaven.engine.util.TableTools;
+import io.deephaven.util.BigDecimalUtils;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+public class BigDecimalChunkedNumericalStats implements ChunkedNumericalStatsKernel {
+    private final static int SCALE =
+            Configuration.getInstance().getIntegerWithDefault("BigDecimalStdOperator.scale", 10);
+
+    private long count = 0;
+
+    private BigDecimal sum = BigDecimal.ZERO;
+    private BigDecimal absSum = BigDecimal.ZERO;
+    private BigDecimal sqrdSum = BigDecimal.ZERO;
+
+    private BigDecimal min = null;
+    private BigDecimal max = null;
+    private BigDecimal absMin = null;
+    private BigDecimal absMax = null;
+
+    @Override
+    public Table processChunks(final RowSet rowSet, final ColumnSource<?> columnSource, boolean usePrev) {
+
+        try (final ChunkSource.GetContext getContext = columnSource.makeGetContext(CHUNK_SIZE)) {
```

Comment:
All of these could be written with less boilerplate using column iterators.

Reply:
I didn't change the short/int/long or float/double, since they collect stats per-chunk to avoid greater loss of precision.

Reply:
Seems like a fair point.